pkg/kv/kvserver/replica_raft.go

```diff
@@ -1464,7 +1466,7 @@ func (r *Replica) refreshProposalsLocked(
 	// already tripped and no probe is running, thus ensuring that even if a
 	// request got added in while the probe was about to shut down, there will
 	// be regular attempts at healing the breaker.
-	if maxSlowProposalDuration > 0 && r.breaker.Signal().Err() == nil {
+	if maxSlowProposalDuration > 0 && r.breaker.Signal().Err() == nil && !destroyed {
```

Comment:
Is it possible to be in this branch on a destroyed replica?

If the replica is destroyed, can we have a spinning behaviour here?
1. `r.breaker.Signal().Err() == nil` leads to tripping the breaker with `r.breaker.TripAsync(err)`
2. `sendProbe` (modified above in this PR) will untrip the breaker
3. repeat

If we can, how do we avoid it?

Reply:
Ah, we won't enter this loop if `destroyed` is true.